packages/expo-updates/ios/EXUpdates/Logging/UpdatesLogReader.swift

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+// Reads expo-updates logs
+
+import Foundation
+import OSLog
+
+import ExpoModulesCore
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+@objc(EXUpdatesLogReader)
+public class UpdatesLogReader: NSObject {
+
+  /**
+   Get expo-updates logs newer than the given date
+   Returns an Objective-C NSArray of strings
+   The strings are all in the JSON format of UpdatesLogEntry
+   */
+  @objc(getLogEntriesNewerThan:)
+  public func getLogEntries(newerThan: Date) -> NSArray {
+    let result = NSMutableArray()
+    do {
+      let logStore = try OSLogStore(scope: .currentProcessIdentifier)
+      // Get all the logs since the given date.
+      let position = logStore.position(date: newerThan)
+
+      // Fetch log objects.
+      let allEntries = try logStore.getEntries(at: position)
+
+      // Filter the log to select our subsystem and the expo-updates category
+      // and extract just the log message strings
+      let allEntriesMessages = allEntries
+          .compactMap { $0 as? OSLogEntryLog }
+          .filter { $0.subsystem == "dev.expo.modules" }
+          .filter { $0.category == "expo-updates" }
+          .compactMap { $0.composedMessage }
```

Comment:
It looks like we're reading all entries into memory and _then_ filtering them; do you think performance is something worth being concerned about here? (Should we put an upper bound on the `newerThan` date, for example?)

Reply:
I was thinking more along the lines of enforcing a limit in this method, just to keep us from shooting ourselves in the foot in the future? But if you would rather keep the flexibility (or just don't think it will be an issue) that's totally fine w me too

Reply:
I did two things here:
- only allow logs to be searched going back 1 day
- added a predicate to the OSLogStore call so we only retrieve the entries we want and don't have to filter after the fact